src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -646,7 +646,10 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 // We can probably do some optimization here. If the selectedCert is returned by the delegate
                 // we can always go ahead and use the certificate to create our credential
                 // (instead of going anonymous as we do here).
-                if (sessionRestartAttempt && cachedCredentialHandle == null && selectedCert != null)
+                if (sessionRestartAttempt &&
```

Comment:
That's how I interpreted it, and I'm hoping that Cristian or David will reinforce that belief :smile:.